CogniteSdk.Types/Common/Identity.cs

```diff
@@ -98,5 +98,29 @@ public override string ToString()
                 return $"{{ ExternalId = \"{ExternalId}\" }}";
             }
         }
+
+        /// <summary>
+        /// Return true if <paramref name="obj"/> is another, identitical Identity.
+        /// </summary>
+        /// <param name="obj">Object to compare</param>
+        /// <returns>True if equal, false otherwise</returns>
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(this, obj)) return true;
+
+            if (!(obj is Identity other)) return false;
+
+            if (Id.HasValue) return Id == other.Id;
+            else return !other.Id.HasValue && ExternalId == other.ExternalId;
```

Comment:
Else not necessary (at least it's not used previously in method so should be consistent)